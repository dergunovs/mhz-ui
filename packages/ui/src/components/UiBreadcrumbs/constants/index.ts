export const LINKS = [
  { url: '/', title: 'Main' },
  { url: '/category', title: 'Categories' },
  { url: '/cpu', title: 'CPU' },
];

export const DEFAULT_COLOR = 'default';
export const DEFAULT_ARIA_LABEL = 'Breadcrumb';
