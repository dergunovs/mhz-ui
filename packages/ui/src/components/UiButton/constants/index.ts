export const DEFAULT_LAYOUT = 'primary';
export const DEFAULT_TYPE = 'button';
export const DEFAULT_ICON = undefined;

export const DEFAULT_SLOT = 'Button label';
export const DEFAULT_ARIA_LABEL = 'Button';
