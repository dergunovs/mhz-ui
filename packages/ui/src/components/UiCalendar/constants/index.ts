import { ICalendarEvent } from '../interface';

export const EVENTS: ICalendarEvent<object>[] = [
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 1, text: 'Text' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 36%, rgb(235, 0, 123) 69%, rgb(251, 185, 0) 100%)',
  },
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 2, text: 'Text 2' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 41%, rgb(235, 0, 123) 71%, rgb(0, 134, 255) 100%)',
  },
];
