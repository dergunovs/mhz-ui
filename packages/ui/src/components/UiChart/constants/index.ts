export const LABELS = [
  '07.07 - 13.07',
  '14.07 - 20.07',
  '21.07 - 27.07',
  '28.07 - 03.08',
  '04.08 - 10.08',
  '11.08 - 17.08',
  '18.08 - 24.08',
];

export const DATASETS = [
  {
    data: [28, 16, 12, 4, 22, 16, 14],
    label: 'Грудь',
    borderColor: '#00CED1',
    backgroundColor: '#00CED1',
  },
  {
    data: [4, 6, 0, 2, 2, 0, 6],
    label: 'Ноги',
    borderColor: '#000080',
    backgroundColor: '#000080',
  },
  {
    data: [22, 18, 8, 10, 25, 18, 16],
    label: 'Плечи',
    borderColor: '#C41E3A',
    backgroundColor: '#C41E3A',
  },
  {
    data: [0, 6, 0, 6, 5, 4, 2],
    label: 'Пресс',
    borderColor: '#32CD32',
    backgroundColor: '#32CD32',
  },
  {
    data: [32, 24, 12, 10, 33, 24, 20],
    label: 'Руки',
    borderColor: '#DA70D6',
    backgroundColor: '#DA70D6',
  },
  {
    data: [16, 18, 8, 12, 26, 22, 20],
    label: 'Спина',
    borderColor: '#FF8C00',
    backgroundColor: '#FF8C00',
  },
];

export const TYPE = 'Line';
export const TITLE = 'Chart title';
