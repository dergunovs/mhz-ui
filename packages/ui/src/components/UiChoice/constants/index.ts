export const options = [1, 2, 3, 4, 5];
export const modelValue = options[2];
export const title = 'Заголовок';
export const isTall = true;
export const isInput = true;
