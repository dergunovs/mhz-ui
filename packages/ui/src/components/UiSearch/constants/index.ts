export const DEBOUNCE_TIME = 300;

export const MODEL_VALUE = '';
export const LANG = 'ru';
export const IS_SUCCESS = true;

export const RESULTS = {
  products: [
    { _id: '1', title: 'AMD Ryzen 7700X' },
    { _id: '2', title: 'AMD Ryzen 7900X' },
  ],
  manufacturers: [{ _id: '1', title: 'AMD' }],
};

export const SEARCH_SCHEME = [
  { type: 'products', labels: ['title'], url: '/products' },
  { type: 'manufacturers', labels: ['title'], url: '/manufacturers' },
];
