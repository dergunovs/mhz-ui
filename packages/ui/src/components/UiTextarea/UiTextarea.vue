<template>
  <textarea
    :value="props.modelValue"
    @input="handleInput($event.target)"
    :disabled="props.isDisabled"
    :class="$style.textarea"
    rows="5"
    data-test="ui-textarea"
    aria-label="textarea"
  ></textarea>
</template>

<script setup lang="ts">
interface IProps {
  modelValue?: string;
  isDisabled?: boolean;
}

const props = defineProps<IProps>();
const emit = defineEmits<{ 'update:modelValue': [value: string] }>();

function handleInput(target: EventTarget | null) {
  emit('update:modelValue', (target as HTMLInputElement).value);
}
</script>

<style module lang="scss">
.textarea {
  width: 100%;
  padding: 16px;
  font-size: 1rem;
  resize: none;
  outline: none;
  border: 1px solid var(--color-gray);
  border-radius: 16px;

  &:hover {
    border: 1px solid var(--color-gray-dark-extra);
  }

  &:focus {
    border: 1px solid var(--color-primary);
  }

  &:disabled {
    color: var(--color-gray-dark-extra);
    background: var(--color-gray-light-extra);
    border-color: var(--color-transparent);
  }

  &::placeholder {
    color: var(--color-gray-dark-extra);
  }
}
</style>
