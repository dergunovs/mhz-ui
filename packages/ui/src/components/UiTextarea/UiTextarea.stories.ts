import { Meta, StoryObj } from '@storybook/vue3';

import { MODEL_VALUE } from './constants';

import { html } from '@/utils';
import { UiTextarea } from '@/components';

const meta: Meta<typeof UiTextarea> = {
  component: UiTextarea,
  args: {
    modelValue: MODEL_VALUE,
  },
  parameters: {
    docs: {
      description: {
        component: '',
      },
    },
  },
};

const argTypes = {};

type Story = StoryObj<typeof UiTextarea>;

export default meta;

export const Primary: Story = {
  render: (args) => ({
    components: { UiTextarea },
    setup: () => ({ args, argTypes }),

    template: html` <UiTextarea v-bind="args" />`,
  }),
  argTypes,
};
