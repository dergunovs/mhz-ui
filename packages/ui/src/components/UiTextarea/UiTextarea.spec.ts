import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import UiTextarea from './UiTextarea.vue';
import { MODEL_VALUE } from './constants';

import { wrapperFactory } from '@/test';

const textarea = dataTest('ui-textarea');

let wrapper: VueWrapper<InstanceType<typeof UiTextarea>>;

beforeEach(() => {
  wrapper = wrapperFactory(UiTextarea, { modelValue: MODEL_VALUE });
});

enableAutoUnmount(afterEach);

describe('UiTextarea', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(UiTextarea)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('handles v-model behaviour', async () => {
    expect((wrapper.find(textarea).element as HTMLInputElement).value).toBe(MODEL_VALUE);

    const newValue = 'New value';

    await wrapper.find(textarea).setValue(newValue);

    expect(wrapper.emitted('update:modelValue')).toHaveLength(1);
    expect(wrapper.emitted('update:modelValue')?.[0]).toEqual([newValue]);
  });

  it('disables textarea by props', async () => {
    expect(wrapper.find(textarea).attributes('disabled')).toBe(undefined);

    await wrapper.setProps({ isDisabled: true });

    expect(wrapper.find(textarea).attributes('disabled')).toBe('');
  });
});
