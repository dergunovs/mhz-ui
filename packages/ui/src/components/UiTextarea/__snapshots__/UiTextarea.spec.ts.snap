// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UiTextarea > matches snapshot 1`] = `"<textarea class="_textarea_116ac4" rows="5" data-test="ui-textarea" aria-label="textarea" value="Text example"></textarea>"`;
